### PR TITLE
Generates webpack bundle analyzer result when builds sigin widget dev bundle

### PR DESCRIPTION
**Pull request by haishengwu-okta**
## Description

- Generates [webpack bundle analyze](https://www.npmjs.com/package/webpack-bundle-analyzer) result in static format (html) when run `yarn build:dev`

## Screenshots

screen shot 2018-10-02 at 1 54 34 pm

### CODE REVIEW

File: buildtools/webpack/plugins.js [hunk 1]
```diff
@@ -5,9 +5,17 @@ const { readFileSync } = require('fs');
...
 const { DefinePlugin, BannerPlugin, IgnorePlugin, optimize } = require('webpack');
 const { some } = require('underscore');
+const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
 
 const UglifyJsPlugin = optimize.UglifyJsPlugin;
 
+function webpackBundleAnalyzer() {
+  return new BundleAnalyzerPlugin({
+    openAnalyzer: false,
+    reportFilename: 'okta-sign-in.html',
```
Comment (santhoshbalakrishnan) [inline buildtools/webpack/plugins.js]: Can we change the filename to be more specific?

Comment (jmelberg-okta) [reply buildtools/webpack/plugins.js]: +1

File: buildtools/webpack/plugins.js [hunk 2]
```diff
@@ -79,7 +87,7 @@ function plugins(options = {}) {
     return [ emptyModule(), uglify(), banner() ];
   }
   // Use DEBUG/development environment w/ console warnings
-  return [ emptyModule(), devMode() ];
+  return [ emptyModule(), devMode(), webpackBundleAnalyzer() ];
```
Comment (jmelberg-okta) [inline buildtools/webpack/plugins.js]: Any reason why this is only generated for dev builds?

Comment (haishengwu-okta) [reply buildtools/webpack/plugins.js]: good question.
1. `release` build generates 4 bundle files which means need further refactor to the build script.
2. I'm not sure if we want to analyze release build at this moment given dev build already gives a lot insights.
2. adds a little time to release build

File: buildtools/webpack/plugins.js (hunk 1 above)
Comment (haishengwu-okta) [reply buildtools/webpack/plugins.js]: `okta-sign-in-bundle-analyzer-result.html` ?

Comment (jmelberg-okta) [reply buildtools/webpack/plugins.js]: We probably don't event need to namespace it, unless we for-see multiple analysis files:

```
bundle-analyzer-result.html
```

File: buildtools/webpack/plugins.js (hunk 2 above)
Comment (geoffgreer-okta) [reply buildtools/webpack/plugins.js]: I've never seen the measurements differ by more than a few percent.

ScaleFT frontend prod build: screenshot_2018-10-02 webpack bundle analyzer 1

ScaleFT frontend dev build: screenshot_2018-10-02 webpack bundle analyzer

**Comment by jmelberg-okta** [comment]
Before merging, can we please add a screenshot and description of this change to the PR description field?

File: buildtools/webpack/plugins.js (hunk 1 above)
Comment (haishengwu-okta) [reply buildtools/webpack/plugins.js]: keep the namespace for now :)

File: buildtools/webpack/plugins.js (hunk 2 above)
Comment (haishengwu-okta) [reply buildtools/webpack/plugins.js]: iirc, the diff is about the https://www.npmjs.com/package/webpack-bundle-analyzer#parsed
when hook to any files

**Comment by haishengwu-okta** [comment]
@jmelberg-okta update the PR description and add screenshot

**Pull request merged**